File: leetcode/easy/Math/cpp/976_Largest Perimeter Triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        
        int pMax = 0;
        sort(nums.begin(), nums.end());
        
        for(int i = 2; i < nums.size(); i++){
            
            int a = nums[i-2];
            int b = nums[i-1];
            int c = nums[i];
            if(a+b > c && a+c > b && b+c > a){
                pMax = max(pMax, a+b+c);
            }
        }
        
        return pMax;
    }
};

File: leetcode/easy/greedy/976_Largest Perimeter Triangle.java
class Solution {
    public int largestPerimeter(int[] nums) {
        
        int ans = 0;
        int n = nums.length;
        Integer[] arr = new Integer[n];
        for(int i = 0; i < n; i++){
            arr[i] = nums[i];
        }
        
        Arrays.sort(arr, Collections.reverseOrder());
        
        for(int i = 0; i < n-2; i++){
            if(arr[i] < arr[i+1] + arr[i+2]){
                ans = Math.max(ans, arr[i] + arr[i+1] + arr[i+2]);
            }    
            
        }
        
        return ans;
    }
}
